### PR TITLE
Gaja/dls 10787/port test to use wiremock

### PR DESCRIPTION
**Pull request by gajalakshmiseetharaman**

### CODE REVIEW

**Comment by ruthdas03** [comment]
Please squash and commit, there are quite a few commits in this branch.

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala [hunk 1]
```diff
@@ -19,63 +19,85 @@ package uk.gov.hmrc.helptosavefrontend.connectors
...
 
-  val helpToSaveUrl = "http://localhost:7001"
+  override def fakeApplication(): Application = new GuiceApplicationBuilder().configure(config).build()
+  lazy val connector: HelpToSaveConnector = app.injector.instanceOf[HelpToSaveConnectorImpl]
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+  implicit lazy val ec: ExecutionContext = app.injector.instanceOf[ExecutionContext]
 
-  val eligibilityURL =
-    s"$helpToSaveUrl/help-to-save/eligibility-check"
+  val eligibilityURL = s"/help-to-save/eligibility-check"
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala:80]: You can remove string interpolation from all these url constants as there is no placeholder in there.

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala [hunk 2]
```diff
@@ -19,63 +19,85 @@ package uk.gov.hmrc.helptosavefrontend.connectors
...
+         |  services {
+         |      help-to-save {
+         |      protocol = http
+         |      host     = $wireMockHost
+         |      port     = $wireMockPort
+         |    }
+         |  }
+         |}
+         |
+         |des {
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala:67]: this config probably doesn't need `des` override. Can remove

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala [hunk 3]
```diff
@@ -247,202 +280,200 @@ class HelpToSaveConnectorSpec extends ControllerSpecWithGuiceApp with HttpSuppor
...
-        mockGet(accountCreateAllowedURL)(Some(HttpResponse(200, Json.toJson(UserCapResponse()), emptyHeaders)))
-
+        val response = HttpResponse(200, Json.toJson(UserCapResponse()), emptyHeaders)
+        when(GET, accountCreateAllowedURL, Map.empty, Map.empty).thenReturn(response.status, response.body)
         await(connector.isAccountCreationAllowed().value) shouldBe Right(UserCapResponse())
       }
+
+      "return an error and the call comes back with any other status other than 200" in {
+        Set(500, 303).foreach { status: Int =>
+          whenever(status =!= 200) {
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala:391]: This check is not needed, would always be the case

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala [hunk 4]
```diff
@@ -247,202 +280,200 @@ class HelpToSaveConnectorSpec extends ControllerSpecWithGuiceApp with HttpSuppor
...
       "be able to handle 200 responses with valid Account json" in {
-        mockGet(url, queryParameters)(Some(HttpResponse(200, Json.toJson(account), emptyHeaders)))
+        val response = HttpResponse(200, Json.toJson(account), emptyHeaders)
+        when(GET, url, queryParameters, Map.empty).thenReturn(response.status, response.body)
         await(connector.getAccount(nino, correlationId).value) shouldBe Right(account)
       }
-
+      "return an error and the call comes back with any other status other than 200" in {
+        Set(500, 303).foreach { status: Int =>
+          whenever(status =!= 200) {
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala:421]: same here, can drop the check for != 200

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveReminderConnectorSpec.scala [hunk 5]
```diff
@@ -59,14 +92,25 @@ class HelpToSaveReminderConnectorSpec extends ControllerSpecWithGuiceApp with Ht
...
 
-      val response =
-        HttpResponse(200, Json.toJson(htsUser), emptyHeaders)
-      mockPost(UpdateHtsURL, Map.empty, htsUser)(Some(response))
+      val response = HttpResponse(200, Json.toJson(htsUser), emptyHeaders)
+
+      when(
+        POST,
+        UpdateHtsURL,
+        Map.empty,
```
Comment (mahesh-hmrc) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveReminderConnectorSpec.scala:100]: Use named parameters, so we can skip passing in empty values for parameters and headers

**Comment by gajalakshmiseetharaman** [comment]
> Please squash and commit, there are quite a few commits in this branch.
@ruthdas03 , squashed now and please verify.

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala (hunk 1 above)
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: Modified.

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala (hunk 2 above)
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: Done

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala (hunk 3 above)
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: Done

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala (hunk 4 above)
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: Done

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveReminderConnectorSpec.scala (hunk 5 above)
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveReminderConnectorSpec.scala]: empty values are removed

File: test/uk/gov/hmrc/helptosavefrontend/connectors/EmailVerificationConnectorSpec.scala [hunk 6]
```diff
@@ -62,55 +101,67 @@ class EmailVerificationConnectorSpec
...
+                .thenReturn(status, "")
               await(connector.verifyEmail(nino, email, name, isNewApplicant)) shouldBe Left(OtherError)
             }
           }
         }
 
         "the future fails" in {
           mockEncrypt(nino + "#" + email)("")
-          mockPost(appConfig.verifyEmailURL, Map.empty[String, String], verificationRequest)(None)
+          when(POST, verifyEmailUrl, body = Some(Json.toJson(verificationRequest).toString())).thenReturn(10, "")
```
Comment (ruthdas03) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/EmailVerificationConnectorSpec.scala:164]: thenReturn has some values here and doesn't replicate case of 'None'.

File: test/uk/gov/hmrc/helptosavefrontend/connectors/IvConnectorSpec.scala [hunk 7]
```diff
@@ -48,26 +82,38 @@ class IvConnectorSpec extends ControllerSpecWithGuiceApp with HttpSupport with S
...
-        result.futureValue should be(Some(IvUnexpectedResponse(httpResponse)))
+        result.futureValue match {
+          case Some(IvUnexpectedResponse(_)) => ()
+          case other                         => fail(s"Expected IvUnexpectedResponse but got $other")
+        }
       }
 
       "handle failure scenarios" in new TestApparatus {
-        mockGet(url)(None)
+        val httpResponse = HttpResponse(0, emptyBody)
```
[the remark this answers is not in the document]
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/IvConnectorSpec.scala]: This case is handled with wireMockServer.stop() and wireMockServer.start()

File: test/uk/gov/hmrc/helptosavefrontend/util/WireMockMethods.scala [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.util
+
```
Comment (ruthdas03) [inline test/uk/gov/hmrc/helptosavefrontend/util/WireMockMethods.scala:18]: Duplicated copyright comment to be removed

**Review by ruthdas03 — CHANGES_REQUESTED** [review]

File: test/uk/gov/hmrc/helptosavefrontend/util/WireMockMethods.scala (hunk 8 above)
Comment (gajalakshmiseetharaman) [reply test/uk/gov/hmrc/helptosavefrontend/util/WireMockMethods.scala:18]: Removed it.

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**